# mayhem_program::instructions::burn::handler  (yours 5288 B at .text+0xfed8; original 5752 B at .text+0xd778)
# distance 312 instructions; evidence: called-by mayhem_program::__private::__global::burn_tokens#1b2c; same-callees 13/13
--- original
+++ yours
@@ -1,203 +1,161 @@
 mov64 r7, r1
-ldxdw r8, [r7+8]
-ldxdw r6, [r8+424]
 mov64 r1, r10
 add64 r1, -336
 call solana_sysvar::clock::<impl solana_sysvar::Sysvar for solana_clock::Clock>::get#06ec
 ldxw r1, [r10-336]
-jeq r1, 1, +0x1508
+jeq r1, 1, +0x1388
 ldxdw r1, [r10-296]
-jsle r6, r1, +0x1a0
-mov64 r1, r10
-add64 r1, -120
+ldxdw r4, [r7+8]
+ldxdw r2, [r4+424]
+jsle r2, r1, +0xe0
+mov64 r1, r10
+add64 r1, -304
 lddw r2, .rodata[05000000]
-call mayhem_program::instructions::pump_amm_interface::pump_swap_types::PumpSwapError::name#252f
-stdw [r10-72], 0
-stdw [r10-80], 1
-stdw [r10-88], 0
-lddw r1, table[0:core::ptr::drop_in_place<alloc::string::String>#75e1, 3:<alloc::string::String as core::fmt::Write>::write_str#ca2d, 4:<alloc::string::String as core::fmt::Write>::write_char#6537, 5:core::fmt::Write::write_fmt#e0dd]
-stxdw [r10-48], r1
-mov64 r1, r10
-add64 r1, -88
-stxdw [r10-56], r1
-lddw r1, 0xe0000020
-stxdw [r10-40], r1
-mov64 r2, r10
-add64 r2, -56
-lddw r1, .rodata[05000000]
-call <mayhem_program::instructions::pump_amm_interface::pump_swap_types::PumpSwapError as core::fmt::Display>::fmt#3dcd
-jne r0, 0, +0x1568
-ldxdw r1, [r10-72]
-stxdw [r10-264], r1
-ldxdw r1, [r10-80]
-stxdw [r10-272], r1
-ldxdw r1, [r10-88]
-stxdw [r10-280], r1
-ldxdw r1, [r10-120]
-stxdw [r10-304], r1
-ldxdw r1, [r10-112]
-stxdw [r10-296], r1
-ldxdw r1, [r10-104]
-stxdw [r10-288], r1
+call mayhem_program::instructions::burn::BurnError::name#3137
+mov64 r1, r10
+add64 r1, -280
+mov64 r2, 5
+call <T as alloc::string::SpecToString>::spec_to_string#e448
 lddw r1, .rodata[70726f6772616d73]
 stxdw [r10-328], r1
-stw [r10-184], 6005
 stb [r10-256], 2
 stw [r10-312], 78
-ja +0x2f8
-ldxb r1, [r8+448]
-jne r1, 2, +0x328
-mov64 r1, r10
-add64 r1, -120
+stdw [r10-320], 48
+stdw [r10-336], 0
+stw [r10-184], 6005
+ja +0x178
+ldxb r1, [r4+448]
+jne r1, 2, +0x198
+mov64 r1, r10
+add64 r1, -304
 lddw r2, .rodata[04000000]
-call mayhem_program::instructions::pump_amm_interface::pump_swap_types::PumpSwapError::name#252f
-stdw [r10-72], 0
-stdw [r10-80], 1
-stdw [r10-88], 0
-lddw r1, table[0:core::ptr::drop_in_place<alloc::string::String>#75e1, 3:<alloc::string::String as core::fmt::Write>::write_str#ca2d, 4:<alloc::string::String as core::fmt::Write>::write_char#6537, 5:core::fmt::Write::write_fmt#e0dd]
-stxdw [r10-48], r1
-mov64 r1, r10
-add64 r1, -88
-stxdw [r10-56], r1
-lddw r1, 0xe0000020
-stxdw [r10-40], r1
-mov64 r2, r10
-add64 r2, -56
-lddw r1, .rodata[04000000]
-call <mayhem_program::instructions::pump_amm_interface::pump_swap_types::PumpSwapError as core::fmt::Display>::fmt#3dcd
-jne r0, 0, +0x1568
-ldxdw r1, [r10-72]
-stxdw [r10-264], r1
-ldxdw r1, [r10-80]
-stxdw [r10-272], r1
-ldxdw r1, [r10-88]
-stxdw [r10-280], r1
-ldxdw r1, [r10-120]
-stxdw [r10-304], r1
-ldxdw r1, [r10-112]
-stxdw [r10-296], r1
-ldxdw r1, [r10-104]
-stxdw [r10-288], r1
+call mayhem_program::instructions::burn::BurnError::name#3137
+mov64 r1, r10
+add64 r1, -280
+mov64 r2, 4
+call <T as alloc::string::SpecToString>::spec_to_string#e448
 lddw r1, .rodata[70726f6772616d73]
 stxdw [r10-328], r1
-stw [r10-184], 6004
 stb [r10-256], 2
 stw [r10-312], 82
 stdw [r10-320], 48
 stdw [r10-336], 0
+stw [r10-184], 6004
 mov64 r1, r10
 add64 r1, -336
 call <anchor_lang::error::Error as core::convert::From<anchor_lang::error::AnchorError>>::from#5ca3
 exit
-ldxdw r1, [r8+232]
+ldxdw r8, [r4+232]
+ldxdw r1, [r4+504]
+ldxdw r1, [r1]
+ldxdw r2, [r1+24]
+stxdw [r10-56], r2
+ldxdw r2, [r1+16]
+stxdw [r10-64], r2
+ldxdw r2, [r1+8]
+stxdw [r10-72], r2
+ldxdw r1, [r1]
+stxdw [r10-80], r1
+ldxdw r2, [r4+88]
+ldxdw r1, [r2+8]
 stxdw [r10-344], r1
-ldxdw r1, [r8+504]
-ldxdw r1, [r1]
-ldxdw r2, [r1+24]
-stxdw [r10-64], r2
-ldxdw r2, [r1+16]
-stxdw [r10-72], r2
-ldxdw r2, [r1+8]
-stxdw [r10-80], r2
-ldxdw r1, [r1]
-stxdw [r10-88], r1
-ldxdw r2, [r8+88]
-ldxdw r4, [r2+8]
-ldxdw r1, [r4]
+ldxdw r1, [r1]
 add64 r1, 1
 mov64 r3, 1
-jeq r1, 0, +0x3c0
+jeq r1, 0, +0x230
 mov64 r3, 0
-ldxdw r5, [r2]
-stxdw [r10-352], r5
-stxdw [r4], r1
-jeq r3, 1, +0x1500
-ldxdw r0, [r2+16]
-ldxdw r1, [r0]
-add64 r1, 1
-mov64 r3, 1
-jeq r1, 0, +0x410
-mov64 r3, 0
-stxdw [r0], r1
-jeq r3, 1, +0x1500
-ldxdw r6, [r8+160]
-ldxdw r5, [r6+8]
+ldxdw r9, [r2]
+ldxdw r5, [r10-344]
+stxdw [r5], r1
+jeq r3, 1, +0x1380
+ldxdw r5, [r2+16]
 ldxdw r1, [r5]
 add64 r1, 1
 mov64 r3, 1
-jeq r1, 0, +0x458
+jeq r1, 0, +0x280
 mov64 r3, 0
-ldxb r9, [r2+42]
-stxdw [r10-384], r9
-ldxb r9, [r2+41]
-stxdw [r10-376], r9
-ldxb r9, [r2+40]
-stxdw [r10-368], r9
-ldxdw r9, [r2+32]
-stxdw [r10-360], r9
+stxdw [r5], r1
+jeq r3, 1, +0x1380
+ldxdw r6, [r4+160]
+ldxdw r1, [r6+8]
+stxdw [r10-352], r1
+ldxdw r1, [r1]
+add64 r1, 1
+mov64 r3, 1
+jeq r1, 0, +0x2d0
+mov64 r3, 0
+ldxb r0, [r2+42]
+stxdw [r10-400], r0
+ldxb r0, [r2+41]
+stxdw [r10-392], r0
+ldxb r0, [r2+40]
+stxdw [r10-384], r0
+ldxdw r0, [r2+32]
+stxdw [r10-376], r0
 ldxdw r2, [r2+24]
-ldxdw r9, [r6]
-stxdw [r5], r1
-jeq r3, 1, +0x1500
-stxdw [r10-400], r2
-stxdw [r10-392], r5
+stxdw [r10-368], r2
+ldxdw r0, [r6]
+ldxdw r2, [r10-352]
+stxdw [r2], r1
+jeq r3, 1, +0x1380
+stxdw [r10-408], r5
 ldxdw r5, [r6+16]
 ldxdw r1, [r5]
 add64 r1, 1
 mov64 r2, 1
-jeq r1, 0, +0x4f8
+jeq r1, 0, +0x378
 mov64 r2, 0
 stxdw [r5], r1
-jeq r2, 1, +0x1500
-stxdw [r10-408], r0
-stxdw [r10-480], r8
-ldxdw r8, [r8+456]
+jeq r2, 1, +0x1380
+stxdw [r10-424], r0
+stxdw [r10-416], r9
+stxdw [r10-360], r8
+stxdw [r10-480], r4
+ldxdw r8, [r4+456]
 ldxdw r0, [r8+8]
 ldxdw r2, [r0]
 add64 r2, 1
 mov64 r3, 1
-jeq r2, 0, +0x550
+jeq r2, 0, +0x3e0
 mov64 r3, 0
 ldxb r1, [r6+42]
 stxdw [r10-440], r1
 ldxb r1, [r6+41]
 stxdw [r10-432], r1
-ldxb r1, [r6+40]
-stxdw [r10-424], r1
-ldxdw r1, [r6+32]
-stxdw [r10-416], r1
+ldxb r9, [r6+40]
+ldxdw r4, [r6+32]
 ldxdw r1, [r6+24]
 ldxdw r6, [r8]
 stxdw [r0], r2
-jeq r3, 1, +0x1500
+jeq r3, 1, +0x1380
 stxdw [r10-448], r6
 ldxdw r2, [r8+16]
 ldxdw r3, [r2]
 add64 r3, 1
 mov64 r6, 1
-jeq r3, 0, +0x5e8
+jeq r3, 0, +0x468
 mov64 r6, 0
 stxdw [r2], r3
-jeq r6, 1, +0x1500
+jeq r6, 1, +0x1380
 ldxdw r3, [r8+24]
 ldxdw r6, [r8+32]
-stxdw [r10-472], r5
-stxdw [r10-464], r9
+stxdw [r10-472], r9
 ldxb r9, [r8+40]
-mov64 r5, r1
+stxdw [r10-456], r5
+stxdw [r10-464], r1
 ldxb r1, [r8+41]
 ldxb r8, [r8+42]
 ldxb r7, [r7+33]
-stxdw [r10-456], r4
+mov64 r5, r4
 mov64 r4, r10
-add64 r4, -1
-stxdw [r10-40], r4
+add64 r4, -113
+stxdw [r10-32], r4
 lddw r4, .rodata[736f6c2d7661756c]
-stxdw [r10-56], r4
-stxb [r10-1], r7
+stxdw [r10-48], r4
+stxb [r10-113], r7
 mov64 r4, r10
-add64 r4, -56
-stxdw [r10-120], r4
+add64 r4, -48
+stxdw [r10-112], r4
 stxb [r10-174], r8
 stxb [r10-175], r1
 stxb [r10-176], r9
@@ -211,46 +169,46 @@
 stxb [r10-222], r1
 ldxdw r1, [r10-432]
 stxb [r10-223], r1
+ldxdw r1, [r10-472]
+stxb [r10-224], r1
+stxdw [r10-232], r5
+ldxdw r1, [r10-464]
+stxdw [r10-240], r1
+ldxdw r1, [r10-456]
+stxdw [r10-248], r1
+ldxdw r1, [r10-352]
+stxdw [r10-256], r1
 ldxdw r1, [r10-424]
-stxb [r10-224], r1
-ldxdw r1, [r10-416]
-stxdw [r10-232], r1
-stxdw [r10-240], r5
-ldxdw r1, [r10-472]
-stxdw [r10-248], r1
+stxdw [r10-264], r1
+ldxdw r1, [r10-400]
+stxb [r10-270], r1
 ldxdw r1, [r10-392]
-stxdw [r10-256], r1
-ldxdw r1, [r10-464]
-stxdw [r10-264], r1
+stxb [r10-271], r1
 ldxdw r1, [r10-384]
-stxb [r10-270], r1
+stxb [r10-272], r1
 ldxdw r1, [r10-376]
-stxb [r10-271], r1
+stxdw [r10-280], r1
 ldxdw r1, [r10-368]
-stxb [r10-272], r1
-ldxdw r1, [r10-360]
-stxdw [r10-280], r1
-ldxdw r1, [r10-400]
 stxdw [r10-288], r1
 ldxdw r1, [r10-408]
 stxdw [r10-296], r1
-ldxdw r1, [r10-456]
+ldxdw r1, [r10-344]
 stxdw [r10-304], r1
-ldxdw r1, [r10-352]
+ldxdw r1, [r10-416]
 stxdw [r10-312], r1
-stdw [r10-32], 1
-stdw [r10-48], 9
-stdw [r10-112], 2
-mov64 r1, r10
-add64 r1, -120
+stdw [r10-24], 1
+stdw [r10-40], 9
+stdw [r10-104], 2
+mov64 r1, r10
+add64 r1, -112
 stxdw [r10-168], r1
-ldxdw r1, [r10-88]
+ldxdw r1, [r10-80]
 stxdw [r10-152], r1
-ldxdw r1, [r10-80]
+ldxdw r1, [r10-72]
 stxdw [r10-144], r1
-ldxdw r1, [r10-72]
+ldxdw r1, [r10-64]
 stxdw [r10-136], r1
-ldxdw r1, [r10-64]
+ldxdw r1, [r10-56]
 stxdw [r10-128], r1
 stdw [r10-320], 0
 stdw [r10-328], 8
@@ -258,54 +216,54 @@
 stdw [r10-160], 1
 mov64 r1, r10
 add64 r1, -336
-ldxdw r8, [r10-344]
+ldxdw r8, [r10-360]
 mov64 r2, r8
 call anchor_spl::token_2022::burn#378a
-jne r0, 2, +0x320
+jne r0, 2, +0x190
 ldxdw r7, [r10-480]
 stxdw [r7+440], r8
 ldxdw r1, [r7+88]
 ldxdw r1, [r1]
 ldxdw r2, [r1+24]
+stxdw [r10-88], r2
+ldxdw r2, [r1+16]
 stxdw [r10-96], r2
+ldxdw r2, [r1+8]
+stxdw [r10-104], r2
+ldxdw r1, [r1]
+stxdw [r10-112], r1
+ldxdw r1, [r7+344]
+ldxdw r1, [r1]
+ldxdw r2, [r1+24]
+stxdw [r10-56], r2
 ldxdw r2, [r1+16]
-stxdw [r10-104], r2
+stxdw [r10-64], r2
 ldxdw r2, [r1+8]
-stxdw [r10-112], r2
-ldxdw r1, [r1]
-stxdw [r10-120], r1
-ldxdw r1, [r7+344]
-ldxdw r1, [r1]
-ldxdw r2, [r1+24]
-stxdw [r10-64], r2
-ldxdw r2, [r1+16]
 stxdw [r10-72], r2
-ldxdw r2, [r1+8]
-stxdw [r10-80], r2
-ldxdw r1, [r1]
-stxdw [r10-88], r1
-mov64 r1, r10
-add64 r1, -56
+ldxdw r1, [r1]
+stxdw [r10-80], r1
+mov64 r1, r10
+add64 r1, -48
 call solana_sysvar::clock::<impl solana_sysvar::Sysvar for solana_clock::Clock>::get#06ec
-ldxw r1, [r10-56]
-jeq r1, 1, +0x15b8
-ldxdw r1, [r10-120]
+ldxw r1, [r10-48]
+jeq r1, 1, +0x13e8
+ldxdw r1, [r10-112]
 stxdw [r10-336], r1
-ldxdw r1, [r10-112]
+ldxdw r1, [r10-104]
 stxdw [r10-328], r1
-ldxdw r1, [r10-104]
+ldxdw r1, [r10-96]
 stxdw [r10-320], r1
-ldxdw r1, [r10-96]
+ldxdw r1, [r10-88]
 stxdw [r10-312], r1
-ldxdw r1, [r10-88]
+ldxdw r1, [r10-80]
 stxdw [r10-304], r1
-ldxdw r1, [r10-80]
+ldxdw r1, [r10-72]
 stxdw [r10-296], r1
-ldxdw r1, [r10-72]
+ldxdw r1, [r10-64]
 stxdw [r10-288], r1
-ldxdw r1, [r10-64]
+ldxdw r1, [r10-56]
 stxdw [r10-280], r1
-ldxdw r6, [r10-16]
+ldxdw r6, [r10-8]
 call _RNvCscR3e5J6VSLN_7___rustc35___rust_no_alloc_shim_is_unstable_v2
 lddw r1, 0x300000000
 ldxdw r2, [r1]
@@ -313,21 +271,21 @@
 mov64 r4, 1
 mov64 r1, r2
 add64 r1, -256
-jle r1, r2, +0xa80
-jeq r4, 0, +0xa90
+jle r1, r2, +0x900
+jeq r4, 0, +0x910
 lddw r1, 0x300007f00
-jne r2, 0, +0xab0
+jne r2, 0, +0x930
 lddw r2, 0x300000007
-jgt r1, r2, +0xad0
-ja +0xff0
+jgt r1, r2, +0x950
+ja +0xe70
 mov64 r4, 0
-jne r4, 0, +0xa48
+jne r4, 0, +0x8c8
 mov64 r3, r1
 lddw r1, 0x300007f00
-jeq r2, 0, +0xa60
+jeq r2, 0, +0x8e0
 mov64 r1, r3
 lddw r2, 0x300000007
-jle r1, r2, +0xff0
+jle r1, r2, +0xe70
 mov64 r9, r10
 add64 r9, -304
 lddw r2, 0x300000000
@@ -352,10 +310,10 @@
 ldxdw r2, [r9+24]
 stxdw [r1+72], r2
 stxdw [r1+80], r6
-stxdw [r10-56], r1
-stdw [r10-48], 88
-mov64 r1, r10
-add64 r1, -56
+stxdw [r10-48], r1
+stdw [r10-40], 88
+mov64 r1, r10
+add64 r1, -48
 mov64 r6, 1
 mov64 r2, 1
 call anchor_lang::solana_program::log::sol_log_data#9398
@@ -363,21 +321,21 @@
 ldxdw r4, [r1+8]
 ldxdw r2, [r4]
 add64 r2, 1
-jeq r2, 0, +0xc08
+jeq r2, 0, +0xa88
 mov64 r6, 0
 ldxdw r8, [r1]
 stxdw [r4], r2
-jeq r6, 1, +0x1500
+jeq r6, 1, +0x1380
 ldxdw r5, [r1+16]
 ldxdw r2, [r5]
 add64 r2, 1
 mov64 r3, 1
-jeq r2, 0, +0xc50
+jeq r2, 0, +0xad0
 mov64 r3, 0
-stxdw [r10-352], r4
+stxdw [r10-344], r4
 stxdw [r5], r2
-jeq r3, 1, +0x1500
-stxdw [r10-360], r5
+jeq r3, 1, +0x1380
+stxdw [r10-352], r5
 ldxb r2, [r1+42]
 stxdw [r10-400], r2
 ldxb r2, [r1+41]
@@ -391,45 +349,45 @@
 ldxdw r1, [r7+88]
 ldxdw r1, [r1]
 ldxdw r2, [r1+24]
+stxdw [r10-88], r2
+ldxdw r2, [r1+16]
 stxdw [r10-96], r2
+ldxdw r2, [r1+8]
+stxdw [r10-104], r2
+ldxdw r1, [r1]
+stxdw [r10-112], r1
+ldxdw r1, [r7+344]
+ldxdw r1, [r1]
+ldxdw r2, [r1+24]
+stxdw [r10-56], r2
 ldxdw r2, [r1+16]
-stxdw [r10-104], r2
+stxdw [r10-64], r2
 ldxdw r2, [r1+8]
-stxdw [r10-112], r2
-ldxdw r1, [r1]
-stxdw [r10-120], r1
-ldxdw r1, [r7+344]
-ldxdw r1, [r1]
-ldxdw r2, [r1+24]
-stxdw [r10-64], r2
-ldxdw r2, [r1+16]
 stxdw [r10-72], r2
-ldxdw r2, [r1+8]
-stxdw [r10-80], r2
-ldxdw r1, [r1]
-stxdw [r10-88], r1
-mov64 r1, r10
-add64 r1, -56
+ldxdw r1, [r1]
+stxdw [r10-80], r1
+mov64 r1, r10
+add64 r1, -48
 call solana_sysvar::clock::<impl solana_sysvar::Sysvar for solana_clock::Clock>::get#06ec
-ldxw r1, [r10-56]
-jeq r1, 1, +0x1618
-ldxdw r1, [r10-120]
+ldxw r1, [r10-48]
+jeq r1, 1, +0x1448
+ldxdw r1, [r10-112]
 stxdw [r10-336], r1
-ldxdw r1, [r10-112]
+ldxdw r1, [r10-104]
 stxdw [r10-328], r1
-ldxdw r1, [r10-104]
+ldxdw r1, [r10-96]
 stxdw [r10-320], r1
-ldxdw r1, [r10-96]
+ldxdw r1, [r10-88]
 stxdw [r10-312], r1
-ldxdw r1, [r10-88]
+ldxdw r1, [r10-80]
 stxdw [r10-304], r1
-ldxdw r1, [r10-80]
+ldxdw r1, [r10-72]
 stxdw [r10-296], r1
-ldxdw r1, [r10-72]
+ldxdw r1, [r10-64]
 stxdw [r10-288], r1
-ldxdw r1, [r10-64]
+ldxdw r1, [r10-56]
 stxdw [r10-280], r1
-ldxdw r6, [r10-16]
+ldxdw r6, [r10-8]
 call _RNvCscR3e5J6VSLN_7___rustc35___rust_no_alloc_shim_is_unstable_v2
 lddw r1, 0x300000000
 ldxdw r1, [r1]
@@ -437,16 +395,16 @@
 mov64 r4, 1
 mov64 r3, r1
 add64 r3, -256
-jle r3, r1, +0xf98
-jeq r4, 0, +0xfa8
+jle r3, r1, +0xe18
+jeq r4, 0, +0xe28
 lddw r7, 0x300007f00
-jne r1, 0, +0xfc8
+jne r1, 0, +0xe48
 stxdw [r10-408], r8
 lddw r1, 0x300000007
-jle r7, r1, +0xff0
+jle r7, r1, +0xe70
 lddw r8, 0x300000000
 stxdw [r8], r7
-ldxdw r1, [r10-344]
+ldxdw r1, [r10-360]
 stxdw [r7+8], r1
 lddw r1, 0xfaee7c52752f5921
 stxdw [r7], r1
@@ -469,19 +427,19 @@
 stxdw [r7+80], r6
 mov64 r1, 0
 mov64 r6, 96
-jsge r6, 0, +0x1018
+jsge r6, 0, +0xe98
 mov64 r2, 96
 lddw r3, panic_location(/home/runner/work/platform-tools/platform-tools/out/rust/library/core/src/iter/traits/iterator.rs:2027:9)
 call alloc::raw_vec::handle_error#4993
 mov64 r4, 0
-jne r4, 0, +0xe60
+jne r4, 0, +0xce0
 mov64 r2, r3
 lddw r7, 0x300007f00
-jeq r1, 0, +0xe78
+jeq r1, 0, +0xcf8
 mov64 r7, r2
 stxdw [r10-408], r8
 lddw r1, 0x300000007
-jgt r7, r1, +0xe98
+jgt r7, r1, +0xd18
 mov64 r1, 1
 mov64 r2, 256
 lddw r3, panic_location(programs/mayhem-program/src/instructions/burn.rs:10:1)
@@ -489,23 +447,23 @@
 call _RNvCscR3e5J6VSLN_7___rustc35___rust_no_alloc_shim_is_unstable_v2
 ldxdw r1, [r8]
 lddw r3, 0x300008000
-jeq r1, 0, +0x1048
+jeq r1, 0, +0xec8
 mov64 r3, r1
 mov64 r2, r3
 sub64 r2, r6
 mov64 r6, 0
 mov64 r1, 1
 mov64 r4, 1
-jle r2, r3, +0x10a0
-jeq r4, 0, +0x10b0
+jle r2, r3, +0xf20
+jeq r4, 0, +0xf30
 lddw r2, 0x300000007
-jgt r6, r2, +0x10d0
-ja +0xf78
+jgt r6, r2, +0xf50
+ja +0xdf8
 mov64 r4, 0
-jne r4, 0, +0x1080
+jne r4, 0, +0xf00
 mov64 r6, r2
 lddw r2, 0x300000007
-jle r6, r2, +0xf78
+jle r6, r2, +0xdf8
 lddw r1, 0x300000000
 stxdw [r1], r6
 lddw r1, 0x1d9acb512ea545e4
@@ -518,7 +476,7 @@
 ldxb r3, [r3]
 stxb [r2+8], r3
 add64 r1, 1
-jne r1, 88, +0x1108
+jne r1, 88, +0xf88
 call _RNvCscR3e5J6VSLN_7___rustc35___rust_no_alloc_shim_is_unstable_v2
 lddw r1, 0x300000000
 ldxdw r2, [r1]
@@ -527,12 +485,12 @@
 mov64 r3, 0
 mov64 r4, 1
 ldxdw r8, [r10-480]
-jle r1, r2, +0x1398
-jeq r4, 0, +0x13a8
+jle r1, r2, +0x1218
+jeq r4, 0, +0x1228
 lddw r1, 0x300007fde
-jne r2, 0, +0x13c8
+jne r2, 0, +0x1248
 lddw r2, 0x300000008
-jlt r1, r2, +0x13e8
+jlt r1, r2, +0x1268
 lddw r2, 0x300000000
 stxdw [r2], r1
 ldxdw r7, [r10-408]
@@ -545,35 +503,35 @@
 stxdw [r1+8], r3
 stxdw [r1], r2
 sth [r1+32], 1
-stxdw [r10-48], r1
-stdw [r10-40], 1
-stdw [r10-56], 1
+stxdw [r10-40], r1
+stdw [r10-32], 1
+stdw [r10-48], 1
 mov64 r9, r10
 add64 r9, -336
 mov64 r5, r10
-add64 r5, -56
+add64 r5, -48
 mov64 r1, r9
 lddw r2, .rodata[052ae5d7a7daa724]
 mov64 r3, r6
 mov64 r4, 96
 call solana_instruction::Instruction::new_with_bytes#33e2
 ldxdw r1, [r10-400]
-stxb [r10-14], r1
+stxb [r10-6], r1
 ldxdw r1, [r10-392]
-stxb [r10-15], r1
+stxb [r10-7], r1
 ldxdw r1, [r10-384]
-stxb [r10-16], r1
+stxb [r10-8], r1
 ldxdw r1, [r10-376]
+stxdw [r10-16], r1
+ldxdw r1, [r10-368]
 stxdw [r10-24], r1
-ldxdw r1, [r10-368]
+ldxdw r1, [r10-352]
 stxdw [r10-32], r1
-ldxdw r1, [r10-360]
+ldxdw r1, [r10-344]
 stxdw [r10-40], r1
-ldxdw r1, [r10-352]
-stxdw [r10-48], r1
-stxdw [r10-56], r7
+stxdw [r10-48], r7
 mov64 r2, r10
-add64 r2, -56
+add64 r2, -48
 mov64 r1, r9
 mov64 r3, 1
 lddw r4, .data.rel.ro+obj
@@ -583,71 +541,64 @@
 mov64 r1, r0
 lsh64 r1, 32
 rsh64 r1, 32
-jne r1, 26, +0x1400
+jne r1, 26, +0x1280
 mov64 r0, 2
-ldxdw r3, [r10-352]
-ja +0x1418
+ldxdw r3, [r10-344]
+ja +0x1298
 mov64 r4, 0
-jne r4, 0, +0x11a0
+jne r4, 0, +0x1020
 mov64 r3, r1
 lddw r1, 0x300007fde
-jeq r2, 0, +0x11b8
+jeq r2, 0, +0x1038
 mov64 r1, r3
 lddw r2, 0x300000008
-jge r1, r2, +0x11d0
+jge r1, r2, +0x1050
 mov64 r1, 1
 mov64 r2, 34
 call alloc::alloc::handle_alloc_error#decd
 mov64 r1, r0
 call <anchor_lang::error::Error as core::convert::From<solana_program_error::ProgramError>>::from#507a
-ldxdw r3, [r10-48]
+ldxdw r3, [r10-40]
 ldxdw r2, [r3]
 add64 r2, -1
 stxdw [r3], r2
-jne r2, 0, +0x1478
+jne r2, 0, +0x12f8
 mov64 r2, r10
-add64 r2, -48
+add64 r2, -40
 mov64 r6, r1
 mov64 r1, r2
 mov64 r7, r0
 call alloc::rc::Rc<T,A>::drop_slow#7561
 mov64 r1, r6
 mov64 r0, r7
-ldxdw r2, [r10-40]
+ldxdw r2, [r10-32]
 ldxdw r3, [r2]
 add64 r3, -1
 stxdw [r2], r3
-jne r3, 0, +0x14e0
+jne r3, 0, +0x1360
 mov64 r2, r10
-add64 r2, -40
+add64 r2, -32
 mov64 r6, r1
 mov64 r1, r2
 mov64 r7, r0
 call alloc::rc::Rc<T,A>::drop_slow#d793
 mov64 r1, r6
 mov64 r0, r7
-jne r0, 2, +0x320
+jne r0, 2, +0x190
 stb [r8+448], 2
 mov64 r0, 2
-ja +0x320
+ja +0x190
 call abort
 ldxdw r1, [r10-332]
-stxdw [r10-56], r1
+stxdw [r10-48], r1
 mov64 r3, r10
-add64 r3, -56
+add64 r3, -48
 lddw r1, str("called `Result::unwrap()` on an `Err` value")
 mov64 r2, 43
 lddw r4, table[3:<solana_program_error::ProgramError as core::fmt::Debug>::fmt#4fcd]
 lddw r5, panic_location(programs/mayhem-program/src/instructions/burn.rs:77:52)
 call core::result::unwrap_failed#5a05
-mov64 r3, r10
-add64 r3, -1
-lddw r1, str("a Display implementation returned an error unexpectedly")
-mov64 r2, 55
-lddw r4, table[3:<core::fmt::Error as core::fmt::Debug>::fmt#aaa5]
-lddw r5, panic_location(/home/runner/work/platform-tools/platform-tools/out/rust/library/alloc/src/string.rs:2830:14)
-call core::result::unwrap_failed#5a05
-ldxdw r1, [r10-52]
+ldxdw r1, [r10-44]
 stxdw [r10-336], r1
 mov64 r3, r10
 add64 r3, -336
@@ -656,7 +607,7 @@
 lddw r4, table[3:<solana_program_error::ProgramError as core::fmt::Debug>::fmt#4fcd]
 lddw r5, panic_location(programs/mayhem-program/src/instructions/burn.rs:104:33)
 call core::result::unwrap_failed#5a05
-ldxdw r1, [r10-52]
+ldxdw r1, [r10-44]
 stxdw [r10-336], r1
 mov64 r3, r10
 add64 r3, -336
